platform-sdk/swirlds-logging-log4j-appender/src/main/java/com/swirlds/logging/log4j/appender/SwirldsLogAppender.java

```diff
@@ -0,0 +1,198 @@
+     * @param filter The filter to apply.
+     * @param layout The layout of log messages.
+     */
+    protected SwirldsLogAppender(final String name, final Filter filter, final Layout<? extends Serializable> layout) {
+        super(name, filter, layout, false, Property.EMPTY_ARRAY);
+    }
+
+    /**
+     * Factory method to create a SwirldsLogAppender instance.
+     * This method is used by the Log4j2 framework.
+     *
+     * @param name   The name of the appender.
+     * @param layout The layout of log messages.
+     * @param filter The filter to apply.
+     *
+     * @return A new instance of SwirldsLogAppender.
+     */
+    @PluginFactory
+    public static SwirldsLogAppender createAppender(
+            @PluginAttribute("name") final String name,
+            @PluginElement("Layout") final Layout<? extends Serializable> layout,
+            @PluginElement("Filters") final Filter filter) {
+        return new SwirldsLogAppender(name, filter, layout);
+    }
+
+    /**
+     * If the log provider was installed from the swirlds-logging API,
+     * {@code event} will be forwarded to the swirlds-logging API.
+     *
+     * @param event The log event to append.
+     * @see SwirldsLogAppender#install(LogEventFactory, LogEventConsumer)
+     */
+    @Override
+    public void append(final LogEvent event) {
+        if (logEventFactory != null && logEventConsumer != null) {
+            logEventConsumer.accept(logEventFactory.createLogEvent(
+                    translateLevel(event.getLevel()),
+                    event.getLoggerName(),
+                    event.getThreadName(),
+                    event.getTimeMillis(),
+                    event.getMessage().getFormattedMessage(),
+                    event.getThrown(),
+                    translateMarker(event.getMarker()),
+                    event.getContextData().toMap()));
+        }
+    }
+
+    /**
+     * Translates Log4j2 markers to swirlds-logging API markers.
+     *
+     * @param marker The Log4j2 marker to translate.
+     *
+     * @return The corresponding swirlds-logging marker.
+     */
+    private Marker translateMarker(@Nullable final org.apache.logging.log4j.Marker marker) {
+        if (marker == null) {
+            return null;
+        }
+
+        final var parents = marker.getParents();
+        if (parents == null || parents.length == 0) {
+            return new Marker(marker.getName());
+        }
+
+        final Marker parent = translateMarker(parents[parents.length - 1]);
+        return new Marker(marker.getName(), parent);
+    }
+
+    /**
+     * Translates Log4j2 log levels to swrirlds-logging API levels.
+     *
+     * @param level The Log4j2 level to translate.
+     *
+     * @return The corresponding swirlds-logging level.
+     */
+    private static Level translateLevel(@NonNull final org.apache.logging.log4j.Level level) {
+        Objects.requireNonNull(level, "level is required");
```

Comment:
that would make it possible that an exception is thrown in log event handling. That should never happen

Reply:
better continue with default value in that case